service/storage_service.cc

```diff
@@ -1001,21 +1001,33 @@ void storage_service::handle_state_normal(inet_address endpoint) {
             (host_id == _gossiper.get_host_id(db().local().get_replace_address().value()))) {
             slogger.warn("Not updating token metadata for {} because I am replacing it", endpoint);
         } else {
+            // Existing node and the node becomes NORMAL status are not the same node
             if (existing && *existing != endpoint) {
+                // Existing node is myself
                 if (*existing == get_broadcast_address()) {
                     slogger.warn("Not updating host ID {} for {} because it's mine", host_id, endpoint);
+                    if (_gossiper.compare_endpoint_startup(endpoint, *existing) > 0) {
+                        slogger.error("Node {} is the new ower of the host ID ={}. I used to be the owner of the host ID. Stop myself to prevent old node to join the cluster!",
+                                endpoint, host_id, get_broadcast_address(), host_id);
+                        stop_transport().get();
+                        abort();
```

Comment:
Why `abort()` and not terminate gracefully?

Reply:
There is no need to gracefully shut down the offending node. Once the transport is shut down, it is isolated from the cluster. I want to to shut down it as soon as possible. Without calling abort(), do you know any alternatives to terminate the process?

Reply:
`abort()` will generate a core dump for no good reason here, which may also take a while to dump if there's lots of ram.

Is it safe to stop a gossiper from a change listener?

We could perhaps signal a `stop_signal` (the one in main.cc) to stop gracefully. It would be the same as interrupting the process.

Reply:
> `abort()` will generate a core dump for no good reason here, which may also take a while to dump if there's lots of ram.
> 
> Is it safe to stop a gossiper from a change listener?

> 
> We could perhaps signal a `stop_signal` (the one in main.cc) to stop gracefully. It would be the same as interrupting the process.

At  the time this patch is written there is no stop_singal thing. This sounds good. Let's discuss in when I summit separately.